### ISSUE TITLE
Support for custom rendered toolbar?

### ISSUE THREAD

**Issue by hugoabernier** [microsoft/roosterjs-react#86]
Thanks for the great work with roosterjs and roosterjs-react!

I was wondering if you're planning on supporting custom-rendered toolbars (or if there is a way to do so currently).

For example, I want to create a toolbar that looks like this:
![image](https://user-images.githubusercontent.com/13972467/55265684-c014e680-524f-11e9-9c05-b671ffd1d34e.png)

I would like to create the toolbar/ribbon myself using Office UI Fabric to get the exact look that I want, and attach the button events perform the actions when pressed/highlight right buttons when a format is selected.

In other words, I need to do more than `buttonRenderer' because I want to change the control that is rendered, not just the icon.

I tried to derive from `RoosterCommandBarPlugin`, but the `FocusOutShell` seems to expect a `RoosterCommandBarPlugin` class, not an interface.

If you could point me in the right direction, I would greatly appreciate it.

**Comment by adriantran** [comment]
@hugoabernier have you tried overriding the buttons?

https://github.com/Microsoft/roosterjs-react/blob/f7fa48e7de9a66a3d7dd7a8191cff8adb208ff86/packages/roosterjs-react-command-bar/lib/schema/RoosterCommandBarSchema.ts#L17

Here it is used in the sample code: https://github.com/Microsoft/roosterjs-react/blob/f7fa48e7de9a66a3d7dd7a8191cff8adb208ff86/sample/script/FocusOutShellSample.tsx#L181

RE creating your own command bar, you don't need to use FocusOutShell to hold the editor and command bar, unless you want to keep the feature where the command bar is hidden unless either the editor or its command bar is focused. If you do use FocusOutShell, just implement your command to use the given calloutClassName value if you have a Callout (such as that heading selection menu, which is inside a Callout.

**Comment by hugoabernier** [comment]
@adrian thanks I'll try what you suggested. I do want to hide the toolbar when the editor or command bar is not focused, and I'm not sure I understand what you mean with "implement your command to use the given calloutClassName value", but I'll dig a bit deeper. Thanks!

**Comment by adriantran** [comment]
They way FocusOutShell works is that if you're focused on elements inside it, the editor and toolbars are "active". When you create a toolbar, you might have a context menu or even a dialog, those live outside of FocusOutShell. If you end up creating your own toolbar, make sure your context menu, dialog, etc. use the calloutClassName given.

**Comment by JiuqingSong** [comment]
This repo is archived now. For react wrapper of roosterjs, please refer to [roosterjs-react](https://github.com/microsoft/roosterjs/tree/master/packages-ui) package in [roosterjs](https://github.com/microsoft/roosterjs) repo. Thanks.
